docs/guides/debugging.md

```diff
@@ -0,0 +1,226 @@
+# Debugging techniques
+
+Errors will happen, be it during builds, unit tests, integration testing, or somewhere else in your pipeline. Earthly provides an interactive debugger to aid debugging command failures.
+
+Debugging an error can be challenging during a traditional image build, since the container state disappears as soon as the build error occurs. A developer may then inject various print commands through out the build to help reason about the state of the system before the failure occurs.
```

Comment:
I would add another sentence saying something along the lines of... Earthly has an interactive mode, which addresses this problem as we will show in this guide.

(Just so the reader doesn't get the impression that there's no solution to this problem and stops reading)